src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,110 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final float GROW_RATE = 2.0f;
+    private int capacity = DEFAULT_CAPACITY;
+    private int threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    private int size;
+    private Node<K, V>[] buckets;
+
+    public MyHashMap() {
+        buckets = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        if (buckets[index] == null) {
+            buckets[index] = new Node<>(key, value);
+            resizeIfThresholdExceeded(size++);
+            return;
+        }
+        Node<K, V> bucket = buckets[index];
+        while (true) {
```

Comment:
You must not use infinite loops when unnecessary. In this particular case, you can create a temporal pointer to the current node and replace it with `Node.next` on every iteration until `Node.next == null`